src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -4726,6 +4736,42 @@ def get_tags(self) -> Union[Dict[str, str], None]:
         # this parameter does not need validation
         return tags
 
+    def get_defender_config(self) -> Union[Dict[str, str], None]:
+        """Obtain the value of defender.
+
+        :return: string or None
+        """
+        disable_defender = self.raw_param.get("disable_defender")
+        if disable_defender:
+            return {"enabled": False, "logAnalyticsWorkspaceResourceId": ""}
+
+        # read the original value passed by the command
+        enable_defender = self.raw_param.get("enable_defender")
+
+        if not enable_defender:
+            return None
+
+        workspace = ""
+        config_file_path = self.raw_param.get("defender_config")
+        if config_file_path:
+            if not os.path.isfile(config_file_path):
+                raise InvalidArgumentValueError(
+                    "{} is not valid file, or not accessable.".format(
+                        config_file_path
+                    )
+                )
+            defender_config = get_file_json(config_file_path)
+            if "logAnalyticsWorkspaceResourceId" in defender_config:
+                workspace = defender_config["logAnalyticsWorkspaceResourceId"]
+
+        if workspace == "":
+            workspace = ensure_default_log_analytics_workspace_for_monitoring(
+                self.cmd,
+                self.get_subscription_id(),
+                self.get_resource_group_name())
```

Comment:
In an update scenario, if the cluster already has defender enabled but user choose to specify the enable option again, without providing the defender config, it seems that this would create a new workspace for the user. Is this expected?